File: 2021/Day 02/aoc02_2.cpp
#include <iostream>
#include <fstream>
using namespace std;

int main() {
	ifstream infile;
	infile.open("aoc02_input.txt");

	int x = 0;
	int y = 0;
	int a = 0;
	string d;
	int s;
	
	while(infile >> d) {
		infile >> s;

		if(d[0] == 'f') {
			x += s;
			y += s * a;
		} else if(d[0] == 'u') {
			a -= s;
		} else if(d[0] == 'd') {
			a += s;
		}
	}
	cout << x * y;

	return 0;
}

File: 2021/Day 02/aoc02_2.py
file = open("aoc02_input.txt")
l = file.read().split("\n")
file.close()

x = 0
y = 0
a = 0
for i in l:
    d, s = i.split()
    s = int(s)

    if d[0] == "f":
        x += s
        y += s * a
    
    if d[0] == "u":
        a -= s
    
    if d[0] == "d":
        a += s

print(x * y)